libc/test/src/search/lfind_test.cpp

```diff
@@ -0,0 +1,46 @@
+//===-- Unittests for lfind -----------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/search/lfind.h"
+#include "test/UnitTest/Test.h"
+
+int compar(void *a, void *b) {
+  return *reinterpret_cast<int *>(a) - *reinterpret_cast<int *>(b);
+}
+
+TEST(LlvmLibcLfindTest, SearchHead) {
+  int list[3] = {1, 2, 3};
+  size_t len = 3;
+  int key = 1;
+  void *ret = LIBC_NAMESPACE::lfind(&key, list, &len, sizeof(int), compar);
+  ASSERT_TRUE(ret == &list[0]);
+}
+
+TEST(LlvmLibcLfindTest, SearchMiddle) {
+  int list[3] = {1, 2, 3};
+  size_t len = 3;
+  int key = 2;
+  void *ret = LIBC_NAMESPACE::lfind(&key, list, &len, sizeof(int), compar);
+  ASSERT_TRUE(ret == &list[1]);
+}
+
+TEST(LlvmLibcLfindTest, SearchTail) {
+  int list[3] = {1, 2, 3};
+  size_t len = 3;
+  int key = 3;
+  void *ret = LIBC_NAMESPACE::lfind(&key, list, &len, sizeof(int), compar);
+  ASSERT_TRUE(ret == &list[2]);
+}
+
+TEST(LlvmLibcLfindTest, SearchNonExistent) {
+  int list[3] = {1, 2, 3};
+  size_t len = 3;
```

Comment:
```suggestion
  constexpr size_t len = 3;
  int list[len] = {1, 2, 3};
```

Reply:
I'll make this constexpr like you suggest.

I'm going to change the name from `len` to `LEN` though since the [style guide](https://libc.llvm.org/dev/code_style.html) suggests `UPPER_SNAKE_CASE` when naming constant declarations.

> const and constexpr variables - They use the capitalized SNAKE_CASE irrespective of whether they are local or global.


We seem to follow that convention in other parts of the code too.
https://github.com/llvm/llvm-project/blob/596fd103f896987463d212921421b3e78b6cf051/libc/src/__support/fixed_point/fx_bits.h#L36

Reply:
It turns out we cannot make `len` constexpr because `lfind` expects a non-const pointer for the `nmemb` parameter, even though it will never be modified.

I mean we could theoretically change `len` to constexpr and then `const_cast` away the constness before passing it to `lfind`, but I don't think that is a good idea.